### ISSUE TITLE
Wrong tree node name

### ISSUE THREAD

**Issue by skartknet** [umco/umbraco-fluidity#26]
It seems that the displayed name is the name of the actual class when the view mode is tree.

I have a very simple setup:
```
public override void Configure(FluidityConfig config)
        {
            config.AddSection("Forms Builder", "icon-brick", sectionConfig =>
            {
                sectionConfig.SetTree("Forms", treeconfig =>
                {
                    treeconfig.AddCollection<Form>(f => f.Id, "Form", "Forms", "Forms list", "icon-paste-in", "icon-layers-alt", collectionConfig =>
                      {
                          collectionConfig.SetNameProperty(f => f.Title);
                          collectionConfig.SetViewMode(Fluidity.FluidityViewMode.Tree);
                          

                          collectionConfig.Editor(editorConfig =>
                          {
                              editorConfig.AddTab("Form", tabConfig =>
                              {                                  
                                  tabConfig.AddField(f => f.Data).SetDataType("Form Builder").SetLabel(String.Empty);
                              });


                              editorConfig.AddTab("Configuration", tabConfig =>
                              {
                              });

                          });


                      });
                });
            });
        }
```

An my class:

```
public class Form
    {
        [PrimaryKeyColumn]
        public int Id { get; set; }
        public string Title { get; set; }
        public string Data { get; set; }
    }
```

It's displaying this:

![image](https://user-images.githubusercontent.com/1685262/37374981-acd13b50-2771-11e8-8b8e-802efe0191bd.png)

**Comment by skartknet** [comment]
It looks like the `CreateEntityTreeNode` method uses just `collection.NameFormat` to set the name of the node. It should use SetNameProperty if set.

**Comment by mattbrailsford** [comment]
Closing as fixed via PR #31
